internal/cmd/login.go

```diff
@@ -437,10 +441,6 @@ func listProviders(client *api.Client) ([]api.Provider, error) {
 }
 
 func promptLoginOptions(client *api.Client) (loginMethod loginMethod, provider *api.Provider, err error) {
-	if isNonInteractiveMode() {
-		return 0, nil, fmt.Errorf("Non-interactive login requires key, instead run: 'infra login SERVER --non-interactive --key KEY")
-	}
```

Comment:
I moved a couple of these to the caller so that we don't have to pass in `options`. 

This was part of making the options local references instead of a static global.